napari/components/command_palette/_api.py

```diff
@@ -0,0 +1,235 @@
+            if tooltip is None:
+                tooltip = getattr(func, "__doc__", "") or ""
+
+            cmd = Command(func, title, desc, tooltip, when)
+            self._commands.append(cmd)
+            return func
+
+        return wrapper if func is None else wrapper(func)
+
+    def add_group(self, title: str) -> CommandGroup:
+        """Add a group to the command palette."""
+        return CommandGroup(title, parent=self)
+
+    def get_widget(self, parent: _QtMainWindow) -> QCommandPalette:
+        """Get a command palette widget for the given parent widget."""
+        from napari._qt.widgets.qt_command_palette import QCommandPalette
+
+        _id = id(parent)
+        if (widget := self._parent_to_palette_map.get(_id)) is None:
+            widget = QCommandPalette(parent)
+            widget.extend_command(self._commands)
+            self._parent_to_palette_map[_id] = widget
+            self._palette_to_parent_map[id(widget)] = parent
+        return widget
+
+    def show_widget(self, parent: _QtMainWindow) -> None:
+        """Show command palette widget."""
+        self.get_widget(parent).show()
+        return None
+
+    def install(self, parent: QtW.QWidget) -> QCommandPalette:
+        """
+        Install command palette on a Qt widget.
+
+        Parameters
+        ----------
+        parent : QtW.QWidget
+            The widget to install on.
+        keys : str, optional
+            If given, this key sequence will be used to show the command palette.
+        """
+        widget = self.get_widget(parent)
+        widget.install_to(parent)
+        return widget
+
+    def update(self, parent: QtW.QWidget | None = None):
+        """Update command palette install to the given parent widget."""
+        if parent is None:
+            for p in self._palette_to_parent_map.values():
+                self.update(p)
+            return None
+        _id = id(parent)
+        widget = self._parent_to_palette_map[_id]
+        widget.clear_commands()
+        widget.extend_command(self._commands)
+        return None
+
+
+class CommandGroup:
+    def __init__(self, title: str, parent: CommandPalette) -> None:
+        self._palette_ref = weakref.ref(parent)
+        self._title = title
+
+    def __repr__(self) -> str:
+        return f"CommandGroup<{self.title}>"
+
+    @property
+    def palette(self) -> CommandPalette:
+        """The parent command palette object."""
+        if palette := self._palette_ref():
+            return palette
+        raise RuntimeError("CommandPalette is already destroyed.")
+
+    @property
+    def title(self) -> str:
+        """The title of this group."""
+        return self._title
+
+    @overload
+    def register(
+        self,
+        func: _F,
+        desc: str | None = None,
+        tooltip: str | None = None,
+        when: Callable[..., bool] | None = None,
+    ) -> _F:
+        ...
+
+    @overload
+    def register(
```

Comment:
Why is it that both `CommandPalette` and `CommandGroup` have the same overloaded `register` methods? Are you ever registered with one but not the other? Can the `CommandGroup` not just be an additional attribute of the `Command`?